colossus/src/main/scala/colossus/streaming/PipeMultiplexer.scala

```diff
@@ -0,0 +1,130 @@
+package colossus
+package streaming
+
+import controller._
+
+trait MultiStream[K, T] extends Stream[T] {
+  def streamId(t: T): K
+}
+
+case class SubSource[K,T](id: K, stream: Source[T])
+case class SubSink[K,T](id: K, stream: Sink[T])
+
+object Multiplexing {
+
+  def multiplex[K,T](base: Sink[T])(implicit ms: MultiStream[K,T]): Sink[SubSource[K,T]] = {
+    val sources = new BufferedPipe[SubSource[K,T]](10)
```

Comment:
Actually I take that back, this doesn't need to be configurable and can have a size of 1.  Items pushed to that pipe are always immediately pulled and backpressure is never applied.